### PR TITLE
Implement dl_iterate_phdr

### LINKED ISSUE #264: rtdl: Implement dl_iterate_phdr()
This function is required for various functionality to work across DSO boundaries, for example stack traces.

Possibility related to the fact that Rust shows empty stack traces on panic.

### PR DESCRIPTION
**Pull request by Geertiebear**
Closes #264.

### CODE REVIEW

File: options/elf/include/link.h [hunk 1]
```diff
@@ -8,13 +8,19 @@ extern "C" {
 #include <elf.h>
 #include <stddef.h>
     
-#define ElfW(type) Elf64_ ## type
+#if defined(__x86_64__) || defined(__aarch64__)
+#	define ElfW(type) Elf64_ ## type
+#elif defined(__i386__)
+#	define ElfW(type) Elf32_ ## type
+#else
+# 	error Unkonwn architecture
```
Comment (avdgrinten) [inline options/elf/include/link.h:16]: Typo.

File: options/rtdl/generic/main.cpp [hunk 2]
```diff
@@ -485,6 +486,29 @@ int __dlapi_reverse(const void *ptr, __dlapi_symbol *info) {
...
+int __dlapi_iterate_phdr(int (*callback)(struct dl_phdr_info *, size_t, void*), void *data) {
+	int last_return = 0;
+	for (auto object : globalScope->_objects) {
+		struct dl_phdr_info info;
+		info.dlpi_addr = object->baseAddress;
+		info.dlpi_name = object->name;
+		info.dlpi_phdr = static_cast<ElfW(Phdr)*>(object->phdrPointer);
+		info.dlpi_phnum = object->phdrCount;
+		info.dlpi_adds = rtsCounter;
+		info.dlpi_subs = 0; // TODO(geert): implement dlclose().
```
Comment (avdgrinten) [inline options/rtdl/generic/main.cpp:499]: `rtsCounter` is not exactly what we want here. As far as I can see, `dlpi_adds` is a global counter that is incremented whenever an object is loaded (?).

**Review by avdgrinten — COMMENTED** [review]
Some comments, remainder LGTM.

File: options/rtdl/generic/main.cpp (hunk 2 above)
Comment (Geertiebear) [reply options/rtdl/generic/main.cpp:499]: Yes, that is ``rtsCounter`` though, right? Or is it also decremented at some point?

Comment (avdgrinten) [reply options/rtdl/generic/main.cpp:499]: Ah, true, I made a mistake there.

**Pull request merged**